### PR TITLE
[K8s] Add user hash to the kind config for multi-user system permission issue

### LINKED ISSUE #3533: [K8S] `sky local up` failed on a multi-user system
On commit `7f30ce5e`

Currently the `sky local up` will create a config yaml under `/tmp` and does not clean it up after local cluster creation. If another user on the same Linux OS tries to run `sky local up`, it will show no permission to write on that fail and abort the creation. Should we cleanup the config or make the name of it user-specific to avoid conflict? Logs:

```bash
Creating local cluster...
To view detailed progress: tail -n100 -f ~/sky_logs/sky-2024-05-09-19-57-51-741127/local_up.log
Failed to create local cluster. Full log: ~/sky_logs/sky-2024-05-09-19-57-51-741127/local_up.log
Error: Traceback (most recent call last):
  File "/home/memory/install/miniconda3/envs/sky/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/memory/install/miniconda3/envs/sky/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/home/memory/skypilot/sky/utils/kubernetes/generate_kind_config.py", line 93, in <module>
    generate_kind_config(args.path, args.port_start, args.port_end,
  File "/home/memory/skypilot/sky/utils/kubernetes/generate_kind_config.py", line 55, in generate_kind_config
    with open(path, 'w', encoding='utf-8') as f:
PermissionError: [Errno 13] Permission denied: '/tmp/skypilot-kind.yaml'
```

### PR DESCRIPTION
**Pull request by cblmemo**
Fixes #3533

Tested (run the relevant ones):

- [ ] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: sky/utils/kubernetes/generate_kind_config.py [hunk 1]
```diff
@@ -71,7 +71,6 @@ def generate_kind_config(path: str,
                                      ' from host to container')
     parser.add_argument('--path',
                         type=str,
-                        default='/tmp/skypilot-kind.yaml',
```
Comment (romilbhardwaj) [inline sky/utils/kubernetes/generate_kind_config.py:74]: Should we leave the default here?

File: sky/cli.py [hunk 2]
```diff
@@ -5097,7 +5097,8 @@ def _deploy_local_cluster(gpus: bool):
 
     # Get directory of script and run it from there
     cwd = os.path.dirname(os.path.abspath(up_script_path))
-    run_command = up_script_path + ' --gpus' if gpus else up_script_path
+    run_command = up_script_path + f' --user {common_utils.get_user_hash()}'
```
Comment (romilbhardwaj) [inline sky/cli.py:5100]: This created ` /tmp/skypilot-kind---user.yaml`. Maybe we want to remove the `--user` here?

```
No kind clusters found.
Generating /tmp/skypilot-kind---user.yaml
D 10-07 22:21:55 skypilot_config.py:228] Using config path: /Users/romilb/.sky/config.yaml
D 10-07 22:21:55 skypilot_config.py:233] Config loaded:
D 10-07 22:21:55 skypilot_config.py:233] {'allowed_clouds': ['kubernetes', 'gcp', 'lambda', 'aws'],
D 10-07 22:21:55 skypilot_config.py:233]  'jobs': {'controller': {'resources': {'cloud': 'kubernetes',
D 10-07 22:21:55 skypilot_config.py:233]                                        'cpus': '4+',
D 10-07 22:21:55 skypilot_config.py:233]                                        'memory': '4+'}}},
D 10-07 22:21:55 skypilot_config.py:233]  'kubernetes': {'provision_timeout': -1},
D 10-07 22:21:55 skypilot_config.py:233]  'serve': {'controller': {'resources': {'cloud': 'kubernetes',
D 10-07 22:21:55 skypilot_config.py:233]                                         'cpus': '4+',
D 10-07 22:21:55 skypilot_config.py:233]                                         'memory': '4+'}}}}
D 10-07 22:21:55 skypilot_config.py:245] Config syntax check passed.
Creating cluster "skypilot" ...
 • Ensuring node image (kindest/node:v1.25.3) 🖼  ...
 ✓ Ensuring node image (kindest/node:v1.25.3) 🖼
 • Preparing nodes 📦   ...
 ✓ Preparing nodes 📦 
 • Writing configuration 📜  ...
 ✓ Writing configuration 📜
 • Starting control-plane 🕹️  ...
 ✓ Starting control-plane 🕹️
 • Installing CNI 🔌  ...
 ✓ Installing CNI 🔌
 • Installing StorageClass 💾  ...
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-skypilot"
You can now use your cluster with:

kubectl cluster-info --context kind-skypilot
```

**Review by romilbhardwaj — COMMENTED** [review]
Thanks @cblmemo!

File: sky/cli.py (hunk 2 above)
Comment (cblmemo) [reply sky/cli.py]: Thanks for pointing it out! I've done some command argument reordering but forgot to test it after. Updated and tested now!

```bash
No kind clusters found.
Generating /tmp/skypilot-kind-4a0782e9.yaml
Creating cluster "skypilot" ...
 • Ensuring node image (kindest/node:v1.27.3) 🖼  ...
 ✓ Ensuring node image (kindest/node:v1.27.3) 🖼
 • Preparing nodes 📦   ...
 ✓ Preparing nodes 📦 
 • Writing configuration 📜  ...
 ✓ Writing configuration 📜
 • Starting control-plane 🕹️  ...
 ✓ Starting control-plane 🕹️
 • Installing CNI 🔌  ...
 ✓ Installing CNI 🔌
 • Installing StorageClass 💾  ...
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-skypilot"
```

**Review by romilbhardwaj — APPROVED** [review]
Thanks @cblmemo!

**Pull request merged**